code/fireball/fireballs.cpp

```diff
@@ -114,12 +107,50 @@ void fireball_play_warphole_close_sound(fireball *fb)
 	snd_play_3d(gamesnd_get_game_sound(sound_index), &fireball_objp->pos, &Eye_position, fireball_objp->radius); // play warp sound effect
 }
 
+static void fireball_generate_unique_id(char *unique_id, int idx)
```

Comment:
This function is unsafe and could cause a buffer-overflow. This should either have a parameter specifying the length of the `char` array or it should use a `SCP_string`.